api/include/opentelemetry/common/kv_properties.h

```diff
@@ -0,0 +1,148 @@
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+#include "opentelemetry/version.h"
+
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace common
+{
+
+// Class to store fixed size array of key-value pairs of string type
+class KeyValueProperties
```

Comment:
Is there reasoning to not use std library containers, e.g. `unordered_map` over `KeyValueProperties`?

Reply:
This is to guarantee ABI stability in api. We don't use std library
 - in api interface as arguments to virtual member functions (https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docs/abi-policy.md)
 - as member variables inside api classes.

Reply:
Thanks, good to know! 🙂